languages/rust/exercises/concept/options/tests/options.rs

```diff
@@ -2,85 +2,92 @@ use options::*;
 
 #[test]
 fn test_reviving_dead_player() {
-    let level: u32 = 34;
     let dead_player = Player {
         health: 0,
         mana: Some(0),
-        level,
+        level: 34,
     };
-    let revived_player = dead_player.revive().expect("reviving a dead player must return Some(player)");
+    let revived_player = dead_player
+        .revive()
+        .expect("reviving a dead player must return Some(player)");
     assert_eq!(revived_player.health, 100);
     assert_eq!(revived_player.mana, Some(100));
-    assert_eq!(revived_player.level, level);
+    assert_eq!(revived_player.level, dead_player.level);
 }
 
 #[test]
 #[ignore]
 fn test_reviving_alive_player() {
-    let level: u32 = 8;
     let alive_player = Player {
         health: 1,
         mana: None,
-        level,
+        level: 8,
     };
     assert_eq!(alive_player.revive().is_none(), true);
 }
 
 #[test]
 #[ignore]
 fn test_cast_spell_with_enough_mana() {
-    let health = 99;
-    let mana = 100;
-    let level = 100;
-    let mana_cost = 3;
+    const HEALTH: u32 = 99;
+    const MANA: u32 = 100;
+    const LEVEL: u32 = 100;
+    const MANA_COST: u32 = 3;
 
     let mut accomplished_wizard = Player {
-        health,
-        mana: Some(mana),
-        level,
+        health: HEALTH,
+        mana: Some(MANA),
+        level: LEVEL,
     };
 
-    assert_eq!(accomplished_wizard.cast_spell(mana_cost), mana_cost * 2);
-    assert_eq!(accomplished_wizard.health, health);
-    assert_eq!(accomplished_wizard.mana, Some(mana - mana_cost));
-    assert_eq!(accomplished_wizard.level, level);
+    assert_eq!(accomplished_wizard.cast_spell(MANA_COST), MANA_COST * 2);
+    assert_eq!(accomplished_wizard.health, HEALTH);
+    assert_eq!(accomplished_wizard.mana, Some(MANA - MANA_COST));
+    assert_eq!(accomplished_wizard.level, LEVEL);
 }
 
 #[test]
 #[ignore]
 fn test_cast_spell_with_insufficient_mana() {
-    let health = 56;
-    let mana = 0;
-    let level = 22;
-    let mana_cost = 3;
-
     let mut no_mana_wizard = Player {
-        health,
-        mana: Some(mana),
-        level,
+        health: 56,
+        mana: Some(2),
+        level: 22,
+    };
+
+    // we want to clone so we can compare before-and-after effects of casting the spell,
+    // but we don't want to introduce that concept to the student yet, so we have to do it manually
+    let clone = Player {
+        health: no_mana_wizard.health,
+        mana: no_mana_wizard.mana,
+        level: no_mana_wizard.level,
```

Comment:
any chance to use `..no_mana_wizard` here, or is that in fact the concept we don't want to introduce yet?